### PR TITLE
💣 Remove powermock

### PR DESCRIPTION
**Pull request by timja**
It doesn't work on Java 17 as far as I can tell, at least with some basic effort, and it appears to be unmaintained.

Mockito can do what it was doing generally.

some tests I couldn't get working or were too painful =/

cc @jglick

### CODE REVIEW

File: core/src/test/java/hudson/FilePathTest.java [hunk 1]
```diff
@@ -478,7 +478,7 @@ private void checkTarUntarRoundTrip(String filePrefix, long fileSize) throws Exc
     }
 
     @Test public void copyToWithPermissionSpecialPermissions() throws IOException, InterruptedException {
-        assumeFalse("Test uses POSIX-specific features", Functions.isWindows());
+        assumeFalse("Test uses POSIX-specific features", Functions.isWindows() || Platform.isDarwin());
```
Comment (timja) [inline core/src/test/java/hudson/FilePathTest.java:481]: fails on my mac, assume fine to skip it there

**Review by jglick — COMMENTED** [review]
Seems like an improvement. I would not mind deleting all the Mockito tests too, but that is another matter.

Recommend also removing the dependency management entry from the plugin POM.

**Comment by MarkEWaite** [comment]
> Recommend also removing the dependency management entry from the plugin POM.

Won't that break plugins using powermock when they upgrade to that parent POM?  Or is that an intentional outcome of what you're proposing so that plugin maintainers are notified that core no longer uses powermock?

**Comment by timja** [comment]
> > Recommend also removing the dependency management entry from the plugin POM.
> 
> Won't that break plugins using powermock when they upgrade to that parent POM? Or is that an intentional outcome of what you're proposing so that plugin maintainers are notified that core no longer uses powermock?

IIRC powermock was originally added to the plugin parent pom to try get people on at least java 11 compatible versions, although it was never really compatible with java 11 you had to add a whole bunch of hacks / workarounds to make it work.

This would be a you should really remove your powermock tests, either migrate to mockito, integration tests, facade pattern or just remove the tests entirely

**Comment by timja** [comment]
I can reproduce the last failure with `mvn clean install` but not `mvn test -Dtest=hudson.model.JobTest` or in my IDE =/

```
[INFO]
[ERROR] Failures:
[ERROR]   JobTest.use_agent_platform_path_separator_when_contribute_path:73 The contributed PATH was not joined using the path separator defined in agent node
Expected: a string containing "/test;"
     but: was "/test:/usr/local/sbin:/Users/timja/.asdf/shims:/usr/local/opt/asdf/libexec/bin:/Users/timja/.nvm/versions/node/v14.15.4/bin:/Users/timja/.rbenv/shims:/usr/local/opt/mysql-client/bin:/usr/local/opt/ruby/bin:/Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home/bin:/Users/timja/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Users/timja/code/jenkins/backend-commit-history-parser/bin:/Users/timja/bin:/usr/local/opt/fzf/bin"
[INFO]
```

**Comment by timja** [comment]
Removal from the plugin pom proposed in https://github.com/jenkinsci/plugin-pom/pull/442, but shouldn't affect this PR

**Review by res0nance — APPROVED** [review]

File: core/pom.xml [hunk 2]
```diff
@@ -418,17 +418,7 @@ THE SOFTWARE.
...
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-module-junit4</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-api-mockito2</artifactId>
+      <artifactId>mockito-inline</artifactId>
       <scope>test</scope>
     </dependency>
```
Comment (basil) [inline core/pom.xml:423]: The corresponding dependencies should be removed from the `<dependencyManagement>` section of the root `pom.xml`, along with the `powermock.version` property in the `<properties>` section.

**Review by basil — APPROVED** [review]

File: core/pom.xml (hunk 2 above)
Comment (timja) [reply core/pom.xml:423]: Can you link to somewhere I missed?

I can't find anything in a search:

```
➜  jenkins git:(nuke-powermock) rg -i powermock
➜  jenkins git:(nuke-powermock) rg -i mockito-core
```

or in https://github.com/jenkinsci/pom/blob/master/pom.xml

**Review by jetersen — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Generally I prefer PowerMock to Mockito. I hope it is a matter of time until PowerMock is adapted though it is likely to be a breaking change too. No objections from me w.r.t this PR, +1

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Pull request merged**